java/vector/src/main/java/org/apache/arrow/vector/ipc/message/ArrowRecordBatch.java

```diff
@@ -232,6 +252,12 @@ public long computeBodyLength() {
       // round up size to the next multiple of 8
       size = DataSizeRoundingUtil.roundUpTo8Multiple(size);
     }
+
+    if (bodyCompression != null) {
```

Comment:
Did you consider making bodyCompression never null (have a static instance that represents zero compression) and making a instance method on it like 'updateMetadataSize(long size)` instead of putting this check here?  Also I'm not familiar with this code but its strange to have to perform this calculation outside of serializing flatbuffers?  Is there something analogous in the C++ code?

Reply:
Sounds reasonable. @rymurr had a similar suggestion.

To make the bodyCompression non-null, we need a default compression codec that makes no compression/decompression. However, our specification does not support such an option now (Please see https://github.com/apache/arrow/blob/master/format/Message.fbs#L45-L53). Providing one would make the implementation not aligning with the specification.

However, we have revised the PR to provide one, because it seems the C++ implementation is not aligning with the specification either (pls see https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/compression.h#L33)

After that, we can guarantee that the bodyCompression is never null, and some other problems also go away.